### PR TITLE
Don't overwrite top level vs code save and undo commands

### PR DESCRIPTION
**Pull request by IanMatthewHuff**
For #4527 

-   [x] Pull request represents a single change (i.e. not fixing disparate/unrelated things in a single PR).
-   [x] Title summarizes what is changing.
-   [x] Has a [news entry](https://github.com/Microsoft/vscode-jupyter/tree/main/news) file (remember to thank yourself!).
-   [ ] Appropriate comments and documentation strings in the code.
-   [ ] Has sufficient logging.
-   [ ] Has telemetry for enhancements.
-   [ ] Unit tests & system/integration tests are added/updated.
-   [ ] [Test plan](https://github.com/Microsoft/vscode-jupyter/blob/main/.github/test_plan.md) is updated as appropriate.
-   [ ] [`package-lock.json`](https://github.com/Microsoft/vscode-jupyter/blob/main/package-lock.json) has been regenerated by running `npm install` (if dependencies have changed).

### CODE REVIEW

File: src/client/datascience/commands/notebookCommands.ts [hunk 1]
```diff
@@ -52,6 +54,16 @@ export class NotebookCommands implements IDisposable {
         }
     }
 
+    private keybindSave() {
+        if (this.notebookEditorProvider.activeEditor) {
+            void this.commandManager.executeCommand('workbench.action.files.save', this.notebookEditorProvider.activeEditor.file);
```
Comment (IanMatthewHuff) [inline src/client/datascience/commands/notebookCommands.ts:59]: Little iffy on if we need to await here. In general given how we handle other VS Code commands I don't believe that we need to. We just send off the save or undo command to VS Code to handle and we don't care what it does or when it does it.

File: package.json [hunk 2]
```diff
@@ -1165,6 +1177,18 @@
                     "category": "Notebook",
                     "when": "notebookEditorFocused"
                 },
+                {
```
Comment (IanMatthewHuff) [inline package.json:1180]: when = false in the commandPalette placement section to hide from command palette. These commands are just for keybindings.

**Comment by IanMatthewHuff** [comment]
Opened for review, but we need to discuss this before commit.

File: package.json [hunk 3]
```diff
@@ -147,15 +147,15 @@
...
+                "command": "jupyter.notebookeditor.keybind.undo"
             },
             {
                 "mac": "S",
                 "win": "S",
                 "linux": "S",
                 "key": "S",
                 "when": "notebookEditorFocused && !inputFocus && notebookViewType == jupyter-notebook && config.jupyter.enableKeyboardShortcuts",
-                "command": "workbench.action.files.save"
+                "command": "jupyter.notebookeditor.keybind.save"
```
Comment (DonJayamanne) [inline package.json:158]: We should file a bug against vS Code, 
the `when` clause makes this PR unnecessary, (we have a clause to ensure this keybinding does not work when a cell has input focus).

Comment (DonJayamanne) [reply package.json:158]: I.e. we're adding a work around for a VS Code bug & I'd prefer VS Code fixes it long term.

**Review by DonJayamanne — CHANGES_REQUESTED** [review]
Please can we create a bug upstream to track this issue.

**Review by DonJayamanne — APPROVED** [review]
Meant to approve it.

File: src/client/datascience/commands/notebookCommands.ts (hunk 1 above)
Comment (DonJayamanne) [reply src/client/datascience/commands/notebookCommands.ts]: Good catch.

**Review by joyceerhl — APPROVED** [review]

**Review by DavidKutu — APPROVED** [review]

**Review by rchiodo — APPROVED** [review]

**Comment by IanMatthewHuff** [comment]
Tests are broken by VS Code insiders notebook API changes. Squashing admin so that I can port to release.

**Pull request merged**